extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/internal/HttpClient.java

```diff
@@ -31,6 +37,46 @@
 @NonNullByDefault
 public class HttpClient {
     private int timeout = 1000;
+    private final Logger logger;
+    private final LinkedList<AsyncPutParameters> commandsQueue;
+    private final Runnable commandExecutor;
+    @Nullable
+    private Future<?> job;
+
+    public HttpClient() {
+        logger = LoggerFactory.getLogger(getClass());
+        commandsQueue = new LinkedList<>();
+
+        commandExecutor = () -> {
+            while (true) {
```

Comment:
The loop is left if a null reference is returned by poll.
At which place null is be added to the queue?

Reply:
As discussed with Kai in the issue (https://github.com/eclipse/smarthome/issues/4794) I've copied this code from the Tradfri binding. (https://github.com/eclipse/smarthome/blob/4204ce06bb28c28e5f711e720f87ef83beff2e27/extensions/binding/org.eclipse.smarthome.binding.tradfri/src/main/java/org/eclipse/smarthome/binding/tradfri/internal/TradfriCoapClient.java)

Can you give me an example how this could be solved in a better way?

Reply:
Perhaps I was not clear...
The http client object is created here: https://github.com/TheNetStriker/smarthome/blob/350ec17ba5218bb650b158a8b2afd0ffc88ef45d/extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/internal/HueBridge.java#L60
At the first call to `putAsync` the command executor is scheduled:
https://github.com/TheNetStriker/smarthome/blob/350ec17ba5218bb650b158a8b2afd0ffc88ef45d/extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/internal/HttpClient.java#L105
Now the endless loop is entered.
My question is: At which place this endless loop is exit (e.g. if a null element has been added to the queue).
I don't identify the logic how the loop is stopped if e.g. the binding is removed at runtime.
If it is e.g. updated isn't there the loop from the old binding dangling and a new one is started, too?

Reply:
Okay, sorry. Now I got it, too :wink: